### DISCUSSION TITLE
Gsoc 2025

### DISCUSSION THREAD

**Discussion by TarunSaxena1996** [JdeRobot/jderobot.github.io#286] [General]
Hi, i am switching my field to Deeplearing and AI, and looking for hands On in opensource projects with is a good combination of robotics and AI, i have a 2 years work experience in Data science field. And looking forward to participate in Gsoc 2025, kindly help me in getting started to secure my position in gsoc 2025.

**Comment by jmplaza** [comment]
Hi @TarunSaxena1996 ,

the best way to start contributing is using and betatesting our software. A good project to contribute to is [RoboticsAcademy ](https://github.com/JdeRobot/RoboticsAcademy), and eventually to submit PRs solving some open issue.

For instance there are two DeepLearning based exercises (Digit Classifier and Human Detection), and we plan to add a third one (End2End visual control of an autonomous car).

I suggest you play with RoboticsAcademy.

**Comment by sergiopaniego** [comment]
Hi @TarunSaxena1996,

Thanks for your interest in contributing to JdeRobot. In addition to RoboticsAcademy, you may also take a look at [DetectionMetrics](https://github.com/JdeRobot/DetectionMetrics). We have a few [`good first issue`](https://github.com/JdeRobot/DetectionMetrics/issues) tagged ideas that you could also contribute.

**Comment by Ravish990** [comment]
Hello,

I am Ravish, currently in my first year at Polaris School of Technology. I have knowledge of Python (including several libraries), Java, JavaScript, HTML, CSS, C++, and basics of the MERN stack. I am eager to contribute to the organization and am confident that I can quickly learn any additional skills required while working on projects.

Open source offers an excellent opportunity to learn by collaborating on various tasks, and I am very excited to get involved. I am ready to contribute my best and give my 100% to the team.

Thank you for considering my enthusiasm and dedication.

**Comment by Ravish990** [comment]
I am in first year of my college mentioned above . I am deeply interested in robotics and eager to contribute to the organization. I believe that engaging in this field will provide invaluable learning opportunities and help me grow significantly.

**Reply by jmplaza** [reply]
Hi @Ravish990 ,

the best way to start contributing is to use our open source software and to submit PRs solving open issues.

1.- JdeRobot supports several software projects. Maybe the most relevant one is RoboticsAcademy, take a look at its repository and its documentation (in the docs directory).

2.- The best channel for discussion is the "Discussions" section on that repo, and creating issues there if needed, commenting them and submitting the corresponding PRs. For long term contributors we have a Slack channel for asynchronous and fast communication.

Cheers,

**Comment by Ravish990** [comment]
@jmplaza 
Thank you, Sir, for the guidance!

I am already checking out the RoboticsAcademy repository and its documentation. I have also joined the Discussions section, so I’ll keep an eye on the open issues and see where I can submit PRs. I am looking forward to engaging with the community and am eagerly excited for the projects and opportunities to learn new things.

Thank you again for your guidance and support and valuable time .

Best regards, Ravish

**Comment by vibhorsachan** [comment]
I hope this message finds  everyone well. My name is Vibhor, and I am a third year Computer Science student specialized in Artificial Intelligence with a keen interest in open-source development and a specialization in Python, Java, and AI engineering. I recently came across your organization’s projects as part of my exploration of Google Summer of Code opportunities and am eager to contribute to your impactful work.

**Reply by jmplaza** [reply]
As previously stated in this thread, the best way to start contributing is to use our software (for instance RoboticsAcademy, BT-studio or VisualCircuit, among others).

**Comment by shubhamachary56821** [comment]
Greeting everyone,
My Name Is Shubham , I am a undergrad Electronics and telecommunication Engg student, completed my diploma and gonna pursue btech(graduation ) this year,  I have these Skills
-**Java, Python and C++**
-**Embedded system programming Using C and microPython**
-**Hands on experience in Raspberrypi 4b and 5, Arduino Uno Nano and Mega,  NodeMCU32, and STM32**
-**Good Communication**
-**Bug Pointing and fixing**
okay, so above are the skills I have , additionally I have worked On a Raspberrypi 4b based Robot which has Obstacle avoidance and self motion mechanism using Ultrasonic sensor based feedback, face Detection + Recognition, Objection detection + Recognition, Speech Module to greet Professor and Students (including guests) , its was I can say kind of humanoid robot, but On wheels , Programmed the whole robot using Python, but later shifted to Java as its lightweight and responsive.
also helped my classmates to build and Code their projects like Arduino based Fire fighting robot, ESp32 based women safety device and arduino nano based MCB Switching device. and more on the list.

In Short I am gonna say that, I am very curious and eager to learn about automation and robotics, If i get a chance on new programs and models I could learn and upskill and at the same time can modify it to work efficiently, but due to limitation of my computational power (under-powered Laptop, Celeron n3350, but equiped with linux) couldn't learn much, but learned many things using web based simulations like WOKWI, TinkerCad and also KiCad on my system.
hope You help me out, and how to contribute to your organizations , trust me I have some good skills, I will make sure to worth every day working with you, and make something better out of it


also I have solved one of your challenges from your website, it was Brownian Motion simulation using python, already posted the answeres and form fillup in that web view , but There is one catch, in the form its written GSOC 2024, instead of gsoc 2025
![Screenshot_2025-03-02_20-31-00](https://github.com/user-attachments/assets/498c9eb2-c0cf-415d-880f-e931483b6756)


if you interested then I can share the solution and the screen recording of the running code (we can say simulation)

**Reply by jmplaza** [reply]
Watch out, we have already updated the Google Form for GSoC-2025

**Reply by shubhamachary56821** [reply]
Okay, I am gonna fill up the section, Btw sir could you please suggest me ,  can we work on JdeRobot projects or repos , do we need linux, or we can work using windows 11 or 10 pc

**Reply by jmplaza** [reply]
Sure, go on with the JdeRobot repos, using the software and solving some issue. Linux is the suggested Operating System, although you can also work on Windows with Docker.

**Comment by pranjal3060** [comment]
Hello everyone,

I'm Pranjal Singh, a fourth-year undergraduate student at IIT Kharagpur, majoring in Electronics and Electrical Comuunication Engineering with a focus on VLSI and Embedded Systems. I've had practical experience as a trainee at Hindustan Aeronautics Limited in the Avionics section.  I have also interned at ABCR Labs in their Tyre Pressure Monitoring System Project. Additionally, I participated in the Kharagpur Winter of Code organized by the Kharagpur Open Source Society in 2023.

I'm proefficient in tools like Vivado and ISE through courses offered in my college, as well as languages such as Python and C++. I'm excited about the opportunity to contribute to the JdeRobot community.
Looking forward to collaborating with all of you!

**Reply by pranjal3060** [reply]
Do we need to contact the project mentors first before filling the Gform for Gsoc 2025 for project discussion

**Reply by jmplaza** [reply]
you can contact the potential mentors before submitting the programming tests, but 'after' is the preferred method. This way (after) your intentions with GSoC-2025 are more solid and convincing. But you can start interacting with them anyway, they will probably require from you to download, install and play with some JdeRobot software (RoboticsAcademy, BTstudio, VisualCircuit...).

**Comment by Aniketsy** [comment]
Hi everyone,

I'm Aniket, a CS student specializing in AI and ML. I'm eager to start contributing here and excited to learn and grow through open-source contributions. I'm particularly interested in the "End-to-End Visual Control of an Autonomous Vehicle Using Deep Learning" project from the GSoC 2025 project list.

As a beginner in open source, I look forward to learning from the community and appreciate any guidance to get started. Excited to collaborate!

Thank you!

**Reply by jmplaza** [reply]
Hi @Aniketsy ,

Using and betatesting the latest release of RoboticsAcademy would be a good way of starting your contributions. Feel free to report your betatesting experience on RoboticsAcademy Discussions channel and RA repository at GitHub.

**Comment by nithika987** [comment]
Hello everyone,

I an Nithika Balaji, and I’m excited to start my journey in open source and contribute to this community. I’m particularly interested in Project #10: Extend DetectionMetrics: GUI, CI Workflow, and Object Detection from the GSOC 2025 project list as it aligns with my passion for computer vision and deep learning.

Even though I’m new to open source, I' m excited to learn, contribute, and interact with the community :)

**Comment by Im-Arth1307** [comment]
Hi everyone,
I am Atharva Jakhetiya, a pre-final year student pursuing Electronics and Communications at Delhi Technological University. I am passionate about Deep Learning and its applications.

I am excited about the opportunity to contribute to Jde Robot and participate in GSOC 2025. I am particularly interested in Project #3 (Robotics-Academy: improve Computer Vision exercises, including with real cameras) and Project #9 (BT-Studio: a tool for programming robots with Behavior Trees).

Over the past few months, I have been actively contributing to open-source projects to enhance my skills in collaborative development. I look forward to engaging with the community, learning from experienced mentors, and making meaningful contributions to Jde Robot.

Thank you

**Reply by jmplaza** [reply]
Hi @Im-Arth1307 , 

Using and betatesting the latest release of BTstudio would be a good way of starting your contributions. Feel free to report your betatesting experience on Btstudio Discussions channel and repository at GitHub.

**Reply by Im-Arth1307** [reply]
@jmplaza yes I'll do my best. Thank you for the advice

**Comment by Im-Arth1307** [comment]
Good evening @jmplaza  sir,
I noticed that the Python (B) challenge for GSoC 2025 hasn’t been updated yet on the JdeRobot website [(link)](https://jderobot.github.io/activities/gsoc/2025#ideas-list). Since the application period begins on March 24, I wanted to check if there are any updates regarding this challenge.
I am very interested in applying to JdeRobot and would appreciate any information on this so that contributors can start preparing their application drafts accordingly.
Thank you for your time and support.
Best regards,
Atharva Jakhetiya

**Comment by Abhinav7609** [comment]
Hi, I am Abhinav. I am a final-year Engineering student at BITS Pilani, specializing in electronics and communication with a minor in data science. I have experience working with hardware accelerators for various deep learning (DL) and large language model (LLM) applications. 

My projects include developing a CNN accelerator on a ZCU 102 FPGA, implementing a Cycle GAN for creating Natural Monet-style paintings, and building an MLP Mixer Vision Model for smaller structured datasets; I have data handling experience using R and Stata for an economics project that involved predicting Entreprenurial success using Machine Learning. I also implemented a 32-bit MIPS processor on FPGA, and currently, I am writing a paper on running Google's BERT on the KV 260 FPGA. 

I am new to open source, looking forward to start contributing especially for project #3 if anyone could help me where to get started :)

**Reply by jmplaza** [reply]
Hi @Abhinav7609 ,

Using and betatesting the latest release of RoboticsAcademy would be a good way of starting your contributions then. Feel free to report your betatesting experience on RoboticsAcademy Discussions channel and repository at GitHub. In particular,  play with the brand new Basic Computer Vision exercise, just using a real camera.

Cheers,
